core/chains/evm/txmgr/orm.go

```diff
@@ -44,17 +46,22 @@ type ORM interface {
 	InsertEthReceipt(receipt *EthReceipt) error
 	InsertEthTx(etx *EthTx) error
 	InsertEthTxAttempt(attempt *EthTxAttempt) error
+	LoadEthTxAttempts(etx *EthTx, qopts ...pg.QOpt) error
+	LoadEthTxesAttempts(etxs []*EthTx, qopts ...pg.QOpt) error
 	MarkAllConfirmedMissingReceipt(chainID big.Int) (err error)
+	MarkOldTxesMissingReceiptAsErrored(blockNum int64, finalityDepth uint32, chainID big.Int, qopts ...pg.QOpt) error
 	PreloadEthTxes(attempts []EthTxAttempt) error
 	SaveConfirmedMissingReceiptAttempt(ctx context.Context, timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error
 	SaveFetchedReceipts(receipts []evmtypes.Receipt, chainID big.Int) (err error)
 	SaveInProgressAttempt(attempt *EthTxAttempt) error
 	SaveInsufficientEthAttempt(timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error
+	SaveReplacementInProgressAttempt(oldAttempt EthTxAttempt, replacementAttempt *EthTxAttempt, qopts ...pg.QOpt) error
 	SaveSentAttempt(timeout time.Duration, attempt *EthTxAttempt, broadcastAt time.Time) error
 	SetBroadcastBeforeBlockNum(blockNum int64, chainID big.Int) error
 	UpdateBroadcastAts(now time.Time, etxIDs []int64) error
 	UpdateEthTxsUnconfirmed(ids []int64) error
 	UpdateEthTxForRebroadcast(etx EthTx, etxAttempt EthTxAttempt) error
+	Close()
```

Comment:
added a Close() method to ORM interface cancel the parent context and all inflight queries. Note that this won't actually cancel all queries YET since some queries accept a context. Once the qopts ticket (mentioned above) is implemented they will. However it's worth knowing that it's possible for a caller to still use the orm even after closing if they pass in their own queryer. I think this is fine but will need to be documented

Reply:
Where is it closed in the application lifecycle?